src/gt4py/next/otf/compilation/build_systems/cmake.py

```diff
@@ -23,7 +24,13 @@
 def get_device_arch() -> str | None:
     if core_defs.CUPY_DEVICE_TYPE == core_defs.DeviceType.CUDA:
         # use `cp` from core_defs to avoid trying to re-import cupy
-        return core_defs.cp.cuda.Device(0).compute_capability  # type: ignore[attr-defined]
+        try:
+            return core_defs.cp.cuda.Device(0).compute_capability  # type: ignore[attr-defined]
+        except core_defs.cp.cuda.runtime.CUDARuntimeError as e:  # type: ignore[attr-defined]
+            warnings.warn(
```

Comment:
Warning looks like

```
/workspace/gt4py/src/gt4py/next/otf/compilation/build_systems/cmake.py:42: UserWarning: Could not determine the CUDA compute capability: cudaErrorInsufficientDriver: CUDA driver version is insufficient for CUDA runtime version
```

Reply:
I would rather not get this warning all the time^^ I think there is nothing to warn in case you don't have a gpu.

Reply:
What's your use-case? I tried this on gitpod (which is my only setup without gpu), and for the tests I didn't even get to the point where this warning was triggered. There was a hard-error before.

Reply:
I pushed one more change maybe that fixes your use-case?